wpilibj/src/main/java/edu/wpi/first/wpilibj/PowerDistribution.java

```diff
@@ -36,27 +54,65 @@ public enum ModuleType {
   }
 
   /**
-   * Constructs a PowerDistribution object.
+   * Constructs a PowerDistribution object. To enable calculating resistance, use
+   * {@link PowerDistribution#PowerDistribution(int, ModuleType, ResistanceCalculator)} instead.
    *
    * @param module The CAN ID of the PDP/PDH.
    * @param moduleType Module type (CTRE or REV).
    */
   public PowerDistribution(int module, ModuleType moduleType) {
-    m_handle = PowerDistributionJNI.initialize(module, moduleType.value);
-    m_module = PowerDistributionJNI.getModuleNumber(m_handle);
-
-    HAL.report(tResourceType.kResourceType_PDP, m_module + 1);
-    SendableRegistry.addLW(this, "PowerDistribution", m_module);
+    this(module, moduleType.value, null);
   }
 
   /**
-   * Constructs a PowerDistribution object.
+   * Constructs a PowerDistribution object. To enable calculating resistance, use
+   * {@link PowerDistribution#PowerDistribution(int, ModuleType, ResistanceCalculator)} instead.
    *
    * <p>Detects the connected PDP/PDH using the default CAN ID (0 for CTRE and 1 for REV).
    */
   public PowerDistribution() {
-    m_handle = PowerDistributionJNI.initialize(kDefaultModule, PowerDistributionJNI.AUTOMATIC_TYPE);
+    this(kDefaultModule, PowerDistributionJNI.AUTOMATIC_TYPE, null);
+  }
+
+  /**
+   * Constructs a PowerDistribution object. To disable calculating resistance, use
+   * {@link PowerDistribution#PowerDistribution(int, ModuleType)} instead.
+   *
+   * @param module The CAN ID of the PDP/PDH.
+   * @param moduleType Module type (CTRE or REV).
+   * @param resistanceCalculator Object to calculate resistance.
+   */
+  public PowerDistribution(
+      int module, ModuleType moduleType, ResistanceCalculator resistanceCalculator) {
+    this(module, moduleType.value, resistanceCalculator);
+  }
+
+  /**
+   * Common constructor to construct a PowerDistribution object.
+   *
+   * @param module The CAN ID of the PDP/PDH.
+   * @param moduleType Module type as an integer (CTRE or REV).
+   * @param resistanceCalculator Object to calculate resistance.
+   */
+  private PowerDistribution(
+      int module, int moduleType, ResistanceCalculator resistanceCalculator) {
```

Comment:
I don't think this is necessary. You could make a ResistanceCalculator instance internally.

Reply:
Do you mean having that private constructor for chaining or having `ResistanceCalculator` be passed in as an argument?

Reply:
You'll need more than one calculator, and those can be created within the constructor itself.

Reply:
Oh, it's for total resistance.

Reply:
If it's the latter, I've added a constructor without parameters for `ResistanceCalculator` now so it's a bit more convenient.

Reply:
But why is that member variable special? The API I'm envisioning doesn't have the user touching ResistanceCalculator at all; it's an implementation detail. All the user does is call getResistance().

Reply:
Oh I see, you want it to be enabled no matter what (I didn't see your previous comments until I came to the files tab)